staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -458,8 +458,8 @@ func (f *DeltaFIFO) listLocked() []interface{} {
 func (f *DeltaFIFO) ListKeys() []string {
 	f.lock.RLock()
 	defer f.lock.RUnlock()
-	list := make([]string, 0, len(f.items))
-	for key := range f.items {
+	list := make([]string, 0, len(f.queue))
+	for _, key := range f.queue {
```

Comment:
This isn't correct change - queue contains only "not yet consumed' items.

Reply:
Hi Wojciech, thanks for your reviews.
Here, we still think the keys in `queue` and `items` are exactly same. Since
1. in the `queueActionLocked` Func, if the key does not exist, it will be added synchronously to `queue` and `items`.
```go
func (f *DeltaFIFO) queueActionLocked(actionType DeltaType, obj interface{}) error {
	...
	if len(newDeltas) > 0 {
		if _, exists := f.items[id]; !exists {
			f.queue = append(f.queue, id)
		}
		f.items[id] = newDeltas
		f.cond.Broadcast()
	} else {
        ...
	}
	return nil
}
```
2. int the `Pop` func, the key in queue and items Will be deleted or re-added synchronously.
```go
func (f *DeltaFIFO) Pop(process PopProcessFunc) (interface{}, error) {
        ...
		id := f.queue[0]
		f.queue = f.queue[1:]
        ...
		delete(f.items, id)
        ...
		err := process(item)
		if e, ok := err.(ErrRequeue); ok {
			f.addIfNotPresent(id, item)
			err = e.Err
		}

	}
}
```
Overall, the keys in `queue` and `items` are always same.
Or did I miss something?

Reply:
/ping @wojtek-t

Reply:
OK - you're right here.